bson/string_codec.go

```diff
@@ -58,12 +57,10 @@ func (sc *stringCodec) decodeType(_ DecodeContext, vr ValueReader, t reflect.Typ
 		if err != nil {
 			return emptyValue, err
 		}
-		if sc.decodeObjectIDAsHex {
+		if sc.decodeObjectIDAsHex || dc.objectIDAsHexString {
```

Comment:
It's unclear to me how `decodeObjectIDAsHex` could ever be set in a non-testing environment. Can we remove this field from `stringCodec`?